uportal-war/src/main/java/org/jasig/portal/portlets/portletadmin/PortletAdministrationHelper.java

```diff
@@ -253,13 +209,71 @@ public PortletDefinitionForm getPortletDefinitionForm(IPerson person, String por
         return form;
     }
 
+    /*
+     * Add groups and permissions to form.
+     */
+    private void getGroupPermissions(IPortletDefinition def, PortletDefinitionForm form, String activity) {
+        final AuthorizationService authService = AuthorizationService.instance();
+        final String portletTargetId = PermissionHelper.permissionTargetIdForPortletDefinition(def);
+
+        /* We are concerned with PORTAL_SUBSCRIBE system */
+        final IPermissionManager pm = authService.newPermissionManager(IPermission.PORTAL_SUBSCRIBE);
+
+        /* Obtain the principals that have permission for the activity on this portlet */
+        final IAuthorizationPrincipal[] principals = pm.getAuthorizedPrincipals(activity, portletTargetId) ;
+        for (IAuthorizationPrincipal principal : principals) {
+            JsonEntityBean bean;
+
+            // first assume this is a group
+            IEntityGroup group = GroupService.findGroup(principal.getKey());
+            if (group != null) {
+                bean = new JsonEntityBean(group, EntityEnum.GROUP);
+            }
+
+            // if a matching group can't be found, try to find a matching
+            // non-group entity
+            else {
```

Comment:
Arrangement of whitespace & comments make these _appear to be_ 2 different if {} clauses.